src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs

```diff
@@ -422,6 +423,30 @@ public void Render(PlotDimensions dims, Bitmap bmp, bool lowQuality = false)
                     DrawLinesWithGaps(points, gfx, penLine);
                 }
 
+                if (DataPointLabels != null)
```

Comment:
I think this should be `DataPointLabels is not null`.

In this case they're the same, but theoretically if we swapped out the type of `DataPointLabels` it is possible we could get different results. And while overloading `==` in such a way as to change the correctness is a little absurd, we could still get (marginally) better performance. If `==` is overloaded the compiler will have to call the overload when we use `==`, rather than just emitting a `ceq` opcode when we use `is`.